nep-18.mediawiki

```diff
@@ -0,0 +1,47 @@
+<pre>
+  NEP: 18
+  Title: Neo Address Resolution
+  Author: Erik Zhang <erik@neo.org>
+  Type: Standard
+  Status: Accepted
+  Created: 2021-02-13
+</pre>
+
+==Abstract==
+
+This NEP describes a standard for storing Neo addresses in TXT records.
+The wallet and client should use the method described in this NEP to store and resolve the TXT record of the domain name, so that users can use the NNS domain name to transfer assets or use dapps.
+
+==Motivation==
+
+NNS currently provides regular DNS service to allow users to add A, CNAME, TXT and AAAA records for domain names.
+However, we do not provide direct support for Neo addresses in NNS.
+We need a way to map domain names in NNS to Neo addresses to support complex application scenarios.
+
+==Specification==
+
+[https://tools.ietf.org/html/rfc1464 RFC 1464] describes the general syntax of a TXT attribute:
+
+<pre>
+"<attribute name>=<attribute value>"
+</pre>
+
+We use <code>address</code> as the attribute name of the Neo address:
+
+<pre>
+"address=<Neo address>"
+</pre>
+
+The following TXT record map the domain name to the Neo address <code>NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY</code>:
+
+<pre>
+"address=NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY"
```

Comment:
Maybe we should specify the address type, like `address.neo=`

Reply:
Do we need to support multiply records for each type?

Reply:
Or separated by ';'

Reply:
But RFC 1464 doesn't allow `;`.

Reply:
I think multiple records is the way to go: if a single record contains a single address, calling by name from inside the contract is much easier (parsing base58 vs splitting a string + parsing). Also multiple records are helpful with other usages for the TXT record.

Reply:
```
neo.xxx.xx TXT "address=NUVPACMnKFhpuHjsRjhUvXz1XhqfGZYVtY"
btc.xxx.xx TXT "address=14qViLJfdGaP4EeHnDyJbEGQysnCpwn1gd"
```